src/Build/Logging/BinaryLogger/BinaryLogRecordKind.cs

```diff
@@ -15,6 +15,7 @@ public enum BinaryLogRecordKind
         EndOfFile = 0,
         BuildStarted,
         BuildFinished,
+        BuildSubmissionStarted,
```

Comment:
The log version needs to be incremented when changing the binary logger contract.

See e.g. this change: https://github.com/dotnet/msbuild/pull/9746/files#diff-0864a6f2e10a87d935fbf19d81370f534e67f33931642dec1f30023b108e2cdc (and other changes in that PR) for referene on adding the new Event type

Reply:
Btw. - to be on a safe side let's add the enum at the end - as the code is converting to enum from bytes being read from the file - old version files then might be misinterpreted

Reply:
Once this is done, you should shortly after update the log view, by updating the BinaryLogRecordKind.cs, BinaryLogger.cs, BuildEventArgsReader.cs and BuildEventArgsWriter.cs in the viewer repo (https://github.com/KirillOsenkov/MSBuildStructuredLog) to be in sync with the changes here.

Sample PR: https://github.com/KirillOsenkov/MSBuildStructuredLog/pull/653

Reply:
I will mark this comment as resolved to keep the PR going. But I'll open a PR to the viewer repo at a later date.